### PR TITLE
added epic option, made visit_group.day validations conditional

### PR DESCRIPTION
**Pull request by ying-pbrc**
When epic is turned off in sparc-request, the day field of the visit_group table is always null. This is preventing the protocol from being imported to sparc-fulfillment. 

I added an optional variable use_epic to our .env and made the validation in app/model/visit_groups conditional. 
 
see details in google group discussion https://groups.google.com/d/msgid/SPARCRequest/xeq59mwrv9vwk5ygad8k2f4g.1441067509476%40email.android.com?utm_medium=email&utm_source=footer

### CODE REVIEW

File: app/models/visit_group.rb [hunk 1]
```diff
@@ -20,7 +20,8 @@ class VisitGroup < ActiveRecord::Base
   validates :arm_id,
             :name,
             presence: true
-  validates :day, presence: true, numericality: true
+  validates :day, presence: true, unless: "ENV.fetch('USE_EPIC'){nil} == 'false'"
```
Comment (amcates) [inline app/models/visit_group.rb]: Wouldn't it make more sense to use:

```
unless: ENV.fetch('USE_EPIC'){false}
```

That would default to false and still work for those with USE_EPIC set to 'true'

Comment (ying-pbrc) [inline app/models/visit_group.rb]: Yep, you are right, it’s simpler. I will make the change.

Comment (ying-pbrc) [inline app/models/visit_group.rb]: Correct me if I am mistaken, there seem to be a syntax error in

unless: ENV.fetch('USE_EPIC'){false}

http://guides.rubyonrails.org/active_record_validations.html#using-a-symbol-with-if-and-unless

I either need to add a new method name use_epic?  Use the :if option with the symbol :use_epic? or use the code I had earlier:
> +  validates :day, presence: true, unless: "ENV.fetch('USE_EPIC'){nil} == 'false'"

I prefer to use the unless option with a string, the default can be set either to nil or false,  and it still work for those with EPIC turned on.

Comment (sparc-request) [inline app/models/visit_group.rb]: It needs to return true or false and not a string. Have you tried it with it turned to true?  It won't interpolate the string unless you use #{}.

On Sep 21, 2015, at 5:43 PM, Ying Wu <notifications@github.com<mailto:notifications@github.com>> wrote:




In app/models/visit_group.rb<https://github.com/sparc-request/sparc-fulfillment/pull/66#discussion_r40030320>:

> @@ -20,7 +20,8 @@ class VisitGroup < ActiveRecord::Base
>    validates :arm_id,
>              :name,
>              presence: true
> -  validates :day, presence: true, numericality: true
> +  validates :day, presence: true, unless: "ENV.fetch('USE_EPIC'){nil} == 'false'"


Correct me if I am mistaken, there seem to be a syntax error in unless: ENV.fetch('USE_EPIC'){false} http://guides.rubyonrails.org/active_record_validations.html#using-a-symbol-with-if-and-unless I either need to add a new method name use_epic? Use the :if option with the symbol :use_epic? or use the code I had earlier:
+ validates :day, presence: true, unless: "ENV.fetch('USE_EPIC'){nil} == 'false'"
I prefer to use the unless option with a string, the default can be set either to nil or false, and it still work for those with EPIC turned on. From: Andrew Cates [mailto:notifications@github.com] Sent: Monday, September 21, 2015 2:54 PM To: sparc-request/sparc-fulfillment Cc: Ying Wu Subject: Re: [sparc-fulfillment] added epic option, made visit_group.day validations conditional (#66<https://github.com/sparc-request/sparc-fulfillment/pull/66>) In app/models/visit_group.rb<https://github.com/sparc-request/sparc-fulfillment/pull/66#discussion_r40017329>:
@@ -20,7 +20,8 @@ class VisitGroup < ActiveRecord::Base validates :arm_id, :name, presence: true - validates :day, presence: true, numericality: true + validates :day, presence: true, unless: "ENV.fetch('USE_EPIC'){nil} == 'false'"
Wouldn't it make more sense to use: unless: ENV.fetch('USE_EPIC'){false} That would default to false and still work for those with USE_EPIC set to 'true' — Reply to this email directly or view it on GitHub<https://github.com/sparc-request/sparc-fulfillment/pull/66/files#r40017329>.

**Pull request closed without merging**